src/math/perfect_cube.rs

```diff
@@ -1,15 +1,8 @@
-pub fn perfect_cube(n: i32) -> bool {
-    // Calculate the cube root using floating-point arithmetic.
-    let val = (n as f64).powf(1.0 / 3.0);
-    // Check if the cube of the cube root equals the original number.
-    (val * val * val) == (n as f64)
-}
-
 // Check if a number is a perfect cube using binary search.
 pub fn perfect_cube_binary_search(n: i64) -> bool {
     // Handle negative numbers, as cube roots are not defined for negatives.
     if n < 0 {
-        return false;
+        return perfect_cube_binary_search(-n);
```

Comment:
Shouldn't it be negative?

Reply:
I am not sure what to you mean.

`n` is a perfect cube if and only if `-n` is a perfect cube. Based on that, the code is correct.

Reply:
My bad, I thought it returned cubic root of the number